lib/Wikia/src/Service/User/Attributes/UserAttributes.php

```diff
@@ -81,35 +85,42 @@ private function loadAttributes( $userId ) {
 	public function setAttribute( $userId, Attribute $attribute ) {
 		$this->loadAttributes( $userId );
 
+		$name = $attribute->getName();
+		$value = $attribute->getValue();
+
 		// If attribute value is null and default exists, set value to default
-		if ( is_null( $attribute->getValue() ) && isset( $this->defaultAttributes[$attribute->getName()] ) ) {
-			$attribute->setValue( $this->defaultAttributes[$attribute->getName()] );
+		if ( is_null( $value ) && isset( $this->defaultAttributes[$name] ) ) {
+			$attribute->setValue( $this->defaultAttributes[$name] );
```

Comment:
So actually, I think this would be fine. Since when we call `getAttribute` for this we'll see the value is `null` and return the default, and when we call `save` well see that the value is the default and skip it.

It still feels a little weird that at times we can have 2 different values for an attribute: the default in `$this->changedAttributes[$name]` and `null` in `$this->attributes[$userId][$name]`. What about not using the local `$value` variable and just using the value from the object?